src/extension.ts

```diff
@@ -75,7 +78,17 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 	// set up the vscode URI handler
 	vscode.window.registerUriHandler({
 		async handleUri(uri:vscode.Uri) {
-			await vscode.commands.executeCommand(extensionFunctions.START_DIDACT_COMMAND, uri);
+			const query = querystring.parse(uri.query);
+			if (query.commandId && query.commandId === extensionFunctions.ADD_TUTORIAL_URI_TO_REGISTRY && query.https && query.name && query.category) {
+				const out : vscode.Uri | undefined = handleVSCodeDidactUriParsingForPath(uri);
+				if (out) {
+					const tutname : string | undefined = getValue(query.name);
+					const tutcat : string | undefined = getValue(query.category);
+					await addNewTutorialWithNameAndCategoryForDidactUri(out, tutname, tutcat);
+				}
```

Comment:
I think we need to add a log in the output channel in case the uri canno tbe parsed

Reply:
when you say `in case the uri cannot be parsed` - are you referring to parsing the query for the incoming uri? or the path to the tutorial itself? in the latter case, since it can be a file or http/https URI, what's the best way to validate it? in the former case, we simply fall through to the older start didact command and pass the uri, so not much validation until we try to open the uri to read the content (and that does add some feedback to the Didact output channel)

Reply:
I had in mind when the `out` variable is `undefined`

Reply:
Added